docs/tutorials/gallery_adding_an_assistant.py

```diff
@@ -0,0 +1,177 @@
+"""
+# Adding an Assistant
+
+Ragna has builtin support for [several assistants][ragna.assistants], but there may be cases where you 
+want to use one that is not currently supported.
+
+This tutorial walks you through the basics of creating an assistant for LLMs that are not
+currently supported. 
+
+!!! note
+
+    This tutorial assumes that our TutorialAssistant class (shown below) is located in the
+    file ragna/assistants/_tutorial.py.
+
+
+"""
+
+# %%
+# ## The Finished Product
+
+# %%
+# !!! note
+#
+#     The code snippet below only includes up to Step 1. To actually include your source
+#     storage in Ragna, you must still perform Step 2.
+
+from typing import Iterator
+
+from ragna.core import Assistant, Source
+
+
+class TutorialAssistant(Assistant):
+    """
+    This is a basic assistant created for didactic purposes
+    """
+
+    def answer(self, prompt: str, sources: list[Source]) -> Iterator[str]:
+        """Answer a prompt given some sources.
+
+        Args:
+            prompt: Prompt to be answered.
+            sources: Sources to use when answering answer the prompt.
+
+        Returns:
+            Answer.
+        """
+        yield self._default_answer(prompt, sources)
```

Comment:
We should definitely mention that you can stream form the assistant by `yield`ing multiple times. Not sure whether we do it as code comment here or as a note down below. We can also link to the [streaming example](https://ragna.chat/en/latest/generated/examples/gallery_streaming/).